src/lib/getKeyDownHandlers.ts

```diff
@@ -0,0 +1,50 @@
+import * as _ from 'lodash'
+import keyboardHandlerFilter from './keyboardHandlerFilter'
+import {
+  IAccessibilityDefinition,
+  AccessibilityActions,
+  ActionsHandler,
+} from 'src/lib/accessibility/interfaces'
+import { IState, IPropsWithVarsAndStyles } from '../../types/theme'
+
+/**
+ * Assigns onKeyDown handler to the Component's part element, based on Component's actions
+ * and keys mappings defined in Accessibility behavior
+ * @param {AccessibilityActions} actions Actions defined in a component.
+ * @param {IAccessibilityDefinition} accessibility The input element which is to loose focus.
+ * @param {IState & IPropsWithVarsAndStyles} props The props which are used to invoke onKeyDown handler passed from top.
+ */
+const getKeyDownHandlers = (
```

Comment:
here also would suggest to do couple of renamings, so that it would be easier to read:

`actions` -> `componentActionHandlers`
`actionsDefinition` -> `behaviorActionDefinitions` (which those, in fact, are)
`elementName` -> `componentPart` (this naming corresponds to the common semantics of anatomy)

`currentActionDef` -> `componentPartActionDefinitions` (as a consequence of previous renamings)
`commonActions` -> `handledActions` (as `common` term rather describes 'how' those are fetched, and not the intent - obtain actions that are supported/handled by component)

this will result in the following code - this one seems to be more readable, as well as its intent is much easier to grasp:

```js
const getKeyDownHandlers = (
  componentActionHandlers: AccessibilityActions,
  accessibility: IAccessibilityDefinition,
  props: IRenderConfigProps,
): ActionsHandler => {
  const keyDownHandlers = {}
  const behaviorActionDefinitions = accessibility.actionsDefinition

  if (!componentActionHandlers || !behaviorActionDefinitions) return keyDownHandlers

  for (const componentPartName in behaviorActionDefinitions) {
    const componentPartActionDefinitions = behaviorActionDefinitions[componentPartName]
    const handledActions = _.intersection(_.keys(componentPartActionDefinitions), _.keys(componentActionHandlers))
    if (!handledActions.length) continue

    keyDownHandlers[componentPartName] = {
      onKeyDown: (event: React.KeyboardEvent) => {
        handledActions.forEach(actionName => {
          const eventHandler = keyboardHandlerFilter(
            componentActionHandlers[actionName],
            componentPartActionDefinitions[actionName].keyCombinations,
          )
          eventHandler && eventHandler(event)
        })

        _.invoke(props, 'onKeyDown', event)
      },
    }
  }

  return keyDownHandlers
}
```

Reply:
please, let me know about your thoughts on that

Reply:
Agree on this suggestions. But referring to comment above, I would make ```componentActionHandlers``` -> ```componentActions```